### DISCUSSION TITLE
UAV Simulation

### DISCUSSION THREAD

**Discussion by whalenJP9731** [JSBSim-Team/jsbsim#415] [Support requests]
Greetings!
I work in the UAV domain. I have come across JSBSim in my search for a physics-based flight model for something that will mimic real world dynamics. However, given that the domain is UAV, most of the commands are abstracted away in the autopilot. Operators will command heading, speed, and altitudes which are interpreted by the onboard autopilot for all surface commands and controls.

Does any such model already exist in the JSB Community? I want to send in basic commands (heading, speed, altitude) and let the autopilot handle everything else. Am I being too naive in what I am hoping to find?

The commands will get sent real time (IE, not through a script)

**Comment by Jaeyoung-Lim** [comment]
PX4 supports SITL(Software-In-The-Loop) simulations through https://github.com/Auterion/px4-jsbsim-bridge

**Comment by seanmcleod** [comment]
Yep, if you want a fairly complete autopilot package then I'd suggest going with something like PX4 or ArduPlane running in SITL mode.

**Comment by whalenJP9731** [comment]
Thank you for the quick responses! I will look into the PX4 link provided. For my general SA, does this repo provide necessary code modifications to JSBSim for simulating UAVs, or example files that are set up to take advantage of already existing features within JSBSim?
